# Belajar cara menggunakan pandas

Pertama-tama kita perlu mengimport pandas, dan kita mengimportanya dengan inisial 'pd'

In [1]:
import pandas as pd

Setelah mengimport pandas, dalam membuat initial data menggunakan pandas kita menggunakan Series dan DataFrame.

Series ialah sekumpulan data. sedangkan DataFrame ialah sekumpulan Series.

Contoh Series

In [4]:
series = pd.Series([44, 70, 23, 12, 33, 54, 21, 78, 88, 23, 43, 45, '44'], name='Score')
series

0     44
1     70
2     23
3     12
4     33
5     54
6     21
7     78
8     88
9     23
10    43
11    45
12    44
dtype: object

Contoh DataFrame

In [7]:
dataframe = pd.DataFrame({
    'name': ['Ersapta', 'Yuda', 'Ayu'],
    'skill': ['Hadoken', 'Sharingan', None]
})

dataframe

name      skill
0  Ersapta    Hadoken
1     Yuda  Sharingan
2      Ayu       None

selain membuat data secara mandiri, kita juga bisa membaca file menggunakan pandas, salah satunya ialah csv.

In [9]:
pandacsv = pd.read_csv('https://raw.githubusercontent.com/FTDS-learning-materials/phase-0/main/src/teams.csv')
pandacsv

division         conference               school_name  \
0     FBS (Division I-A Teams)  American Athletic                Cincinnati   
1     FBS (Division I-A Teams)  American Athletic               Connecticut   
2     FBS (Division I-A Teams)  American Athletic                   Houston   
3     FBS (Division I-A Teams)  American Athletic                Louisville   
4     FBS (Division I-A Teams)  American Athletic                   Memphis   
..                         ...                ...                       ...   
247  FCS (Division I-AA Teams)               SWAC             Jackson State   
248  FCS (Division I-AA Teams)               SWAC  Mississippi Valley State   
249  FCS (Division I-AA Teams)               SWAC          Prairie View A&M   
250  FCS (Division I-AA Teams)               SWAC       Southern University   
251  FCS (Division I-AA Teams)               SWAC            Texas Southern   

                                          roster_url   id  
0    http://espn.go.com/ncf/teams/roster?teamId=2132    1  
1      http://espn.go.com/ncf/teams/roster?teamId=41    2  
2     http://espn.go.com/ncf/teams/roster?teamId=248    3  
3      http://espn.go.com/ncf/teams/roster?teamId=97    4  
4     http://espn.go.com/ncf/teams/roster?teamId=235    5  
..                                               ...  ...  
247  http://espn.go.com/ncf/teams/roster?teamId=2296  248  
248  http://espn.go.com/ncf/teams/roster?teamId=2400  249  
249  http://espn.go.com/ncf/teams/roster?teamId=2504  250  
250  http://espn.go.com/ncf/teams/roster?teamId=2582  251  
251  http://espn.go.com/ncf/teams/roster?teamId=2640  252  

[252 rows x 5 columns]

contoh menggunakan pandacsv.head() yang secara default menampilkan 5 data terdepan

In [6]:
pandacsv.head()

name  main_category  \
0      EIO New Born Baby Gift Set -13 Piecess (Pink)  kids' fashion   
1  exciting Lives - Vintage Motorbike Pocket Watc...  kids' fashion   
2  Jewel string by Bk arts Evil Eye Baby Nazaria ...  kids' fashion   
3                    Superminis Unisex-baby's Bootie  kids' fashion   
4  My Newborn Baby Mittens Caps Gloves and Booty ...  kids' fashion   

   sub_category                                              image  \
0  Baby Fashion  https://m.media-amazon.com/images/I/61X3IMp9dv...   
1  Baby Fashion  https://m.media-amazon.com/images/I/71f3rkWNAF...   
2  Baby Fashion  https://m.media-amazon.com/images/I/51Fc5epiN1...   
3  Baby Fashion  https://m.media-amazon.com/images/I/71MNqAzGnA...   
4  Baby Fashion  https://m.media-amazon.com/images/I/61CHRBjLTd...   

                                                link ratings no_of_ratings  \
0  https://www.amazon.in/EIO-Kidi-Born-Babys-Piec...     4.2         2,958   
1  https://www.amazon.in/exciting-Lives-Motorbike...     4.2           161   
2  https://www.amazon.in/jewel-string-evil-eye-na...     4.2           962   
3  https://www.amazon.in/Superminis-Velvet-Bootie...     4.1         2,046   
4  https://www.amazon.in/My-NewBorn-Baby-Mitten-B...     3.8         5,601   

  discount_price actual_price  
0           ₹649       ₹1,499  
1           ₹485         ₹699  
2           ₹249         ₹799  
3           ₹350         ₹899  
4           ₹299         ₹900

contoh menggunakan pandacsv.head() menampilkan data n terdepan, dimana n adalah jumlah data yang ingin ditampilkan

In [7]:
pandacsv.head(n= 10)

name  main_category  \
0      EIO New Born Baby Gift Set -13 Piecess (Pink)  kids' fashion   
1  exciting Lives - Vintage Motorbike Pocket Watc...  kids' fashion   
2  Jewel string by Bk arts Evil Eye Baby Nazaria ...  kids' fashion   
3                    Superminis Unisex-baby's Bootie  kids' fashion   
4  My Newborn Baby Mittens Caps Gloves and Booty ...  kids' fashion   
5  DHRUVS COLLECTION 92.5 Pure Silver Black Beads...  kids' fashion   
6  Bold N Elegant - Be Bold Inside & Elegant Outs...  kids' fashion   
7            NPRC Unisex Wool Cap (436_Multicolor_M)  kids' fashion   
8          SooWoo® Boy’s Anti-Skid Grip Cotton Socks  kids' fashion   
9  Coolz Kids Chu-Chu Sound Musical First Walking...  kids' fashion   

   sub_category                                              image  \
0  Baby Fashion  https://m.media-amazon.com/images/I/61X3IMp9dv...   
1  Baby Fashion  https://m.media-amazon.com/images/I/71f3rkWNAF...   
2  Baby Fashion  https://m.media-amazon.com/images/I/51Fc5epiN1...   
3  Baby Fashion  https://m.media-amazon.com/images/I/71MNqAzGnA...   
4  Baby Fashion  https://m.media-amazon.com/images/I/61CHRBjLTd...   
5  Baby Fashion  https://m.media-amazon.com/images/I/618KZ5Vlo0...   
6  Baby Fashion  https://m.media-amazon.com/images/I/512Fau61yG...   
7  Baby Fashion  https://m.media-amazon.com/images/I/A1hUaXW-Hm...   
8  Baby Fashion  https://m.media-amazon.com/images/I/81mJmWUJT+...   
9  Baby Fashion  https://m.media-amazon.com/images/I/810iMPEu00...   

                                                link ratings no_of_ratings  \
0  https://www.amazon.in/EIO-Kidi-Born-Babys-Piec...     4.2         2,958   
1  https://www.amazon.in/exciting-Lives-Motorbike...     4.2           161   
2  https://www.amazon.in/jewel-string-evil-eye-na...     4.2           962   
3  https://www.amazon.in/Superminis-Velvet-Bootie...     4.1         2,046   
4  https://www.amazon.in/My-NewBorn-Baby-Mitten-B...     3.8         5,601   
5  https://www.amazon.in/DHRUVS-COLLECTION-Sterli...     4.0           264   
6  https://www.amazon.in/Bold-Elegant-Outside-Kne...     3.9         4,050   
7  https://www.amazon.in/NPRC-Woollen-Monkey-Colo...     4.1             2   
8  https://www.amazon.in/SooWoo%C2%AE-Anti-Skid-C...     NaN           NaN   
9  https://www.amazon.in/Coolz-Chu-Chu-Musical-Wa...     4.2         5,914   

  discount_price actual_price  
0           ₹649       ₹1,499  
1           ₹485         ₹699  
2           ₹249         ₹799  
3           ₹350         ₹899  
4           ₹299         ₹900  
5           ₹440       ₹1,599  
6           ₹549       ₹1,999  
7           ₹149         ₹499  
8           ₹699       ₹1,299  
9           ₹349         ₹499

contoh menggunakan pandacsv.tail() yang secara default menampilkan 5 data terakhir

In [ ]:
pandacsv.tail()

contoh menggunakan pandacsv.tail() menampilkan data n terakhir, dimana n adalah jumlah data yang ingin ditampilkan

In [ ]:
pandacsv.tail(n= 10)

contoh penggunaan pandacsv.shape, yang dimana menampilkan bentuk dari data yang diload dengan format (row, column)

In [8]:
pandacsv.shape

(2208, 9)

disini diketahui data yang kita miliki ialah 2208 baris dan 9 kolom

contoh penggunaan pandacsv.info(), yang dimana menampilkan kesimpulan dari data yang sudah diload, dan kita bisa mengetahui tipe data serta data non null dari masing-masing kolom

In [9]:
pandacsv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2208 entries, 0 to 2207
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   name            2208 non-null   object
 1   main_category   2208 non-null   object
 2   sub_category    2208 non-null   object
 3   image           2208 non-null   object
 4   link            2208 non-null   object
 5   ratings         1262 non-null   object
 6   no_of_ratings   1262 non-null   object
 7   discount_price  2032 non-null   object
 8   actual_price    2152 non-null   object
dtypes: object(9)
memory usage: 155.4+ KB


contoh penggunaan pandacsv.describe(), yang dimana menampilkan hasil dari dasar statistik seperti count dan unique

In [10]:
pandacsv.describe()

name  main_category  \
count                                                2208           2208   
unique                                               2101              1   
top     ARVESA Monthly Birthday Milestone Special Unis...  kids' fashion   
freq                                                   19           2208   

        sub_category                                              image  \
count           2208                                               2208   
unique             1                                               2146   
top     Baby Fashion  https://m.media-amazon.com/images/I/81t2m239-L...   
freq            2208                                                  5   

                                                     link ratings  \
count                                                2208    1262   
unique                                               2208      36   
top     https://www.amazon.in/EIO-Kidi-Born-Babys-Piec...     5.0   
freq                                                    1     150   

       no_of_ratings discount_price actual_price  
count           1262           2032         2152  
unique           259            490          347  
top                1           ₹499         ₹999  
freq             182            136          310

contoh penggunaan pandacsv.columns(), yang dimana menampilkan seluruh nama kolom pada data yang sudah diload

In [11]:
pandacsv.columns

Index(['name', 'main_category', 'sub_category', 'image', 'link', 'ratings',
       'no_of_ratings', 'discount_price', 'actual_price'],
      dtype='object')

contoh penggunaan pandacsv.index(), yang dimana menampilkan info tentang index data yang diload, info start index, index akhirnya serta step per index

In [12]:
pandacsv.index

RangeIndex(start=0, stop=2208, step=1)

contoh penggunaan pandacsv.dtypes(), yang dimana menampilkan seluruh nama kolom beserta tipe data isian kolom tersebut

In [13]:
pandacsv.dtypes

name              object
main_category     object
sub_category      object
image             object
link              object
ratings           object
no_of_ratings     object
discount_price    object
actual_price      object
dtype: object

contoh penggunaan pandacsv.isnull(), yang dimana menghasilkan boolean value dari masing masing data apakah null atau tidak

In [14]:
pandacsv.isnull()

name  main_category  sub_category  image   link  ratings  \
0     False          False         False  False  False    False   
1     False          False         False  False  False    False   
2     False          False         False  False  False    False   
3     False          False         False  False  False    False   
4     False          False         False  False  False    False   
...     ...            ...           ...    ...    ...      ...   
2203  False          False         False  False  False     True   
2204  False          False         False  False  False     True   
2205  False          False         False  False  False    False   
2206  False          False         False  False  False     True   
2207  False          False         False  False  False    False   

      no_of_ratings  discount_price  actual_price  
0             False           False         False  
1             False           False         False  
2             False           False         False  
3             False           False         False  
4             False           False         False  
...             ...             ...           ...  
2203           True           False         False  
2204           True           False         False  
2205          False            True         False  
2206           True           False         False  
2207          False           False         False  

[2208 rows x 9 columns]

pandacsv.nunique()

In [15]:
pandacsv.nunique()

name              2101
main_category        1
sub_category         1
image             2146
link              2208
ratings             36
no_of_ratings      259
discount_price     490
actual_price       347
dtype: int64

contoh penggunaan pandacsv.value_counts(), yang dimana menampilkan data serta total data tersebut

In [16]:
pandacsv.value_counts()

name                                                                                                                           main_category  sub_category  image                                                                                             link                                                                                                                                                                                     ratings  no_of_ratings  discount_price  actual_price
100% Organic Bamboo Cotton Unisex Solid Baby Pants Plain Pajama Solid Lower Leggings Newborn Pajami Diaper Fit Without Pri...  kids' fashion  Baby Fashion  https://m.media-amazon.com/images/I/51txaod+exL._AC_UL320_.jpg                                    https://www.amazon.in/Organic-Bamboo-Leggings-Newborn-Without/dp/B0BQ429KH9/ref=sr_1_9470?qid=1679221542&s=apparel&sr=1-9470                                                             4.4      3              ₹349            ₹699            1
Neobaby Uni

In [17]:
pandacsv['ratings'].value_counts()

5.0    150
4.0    145
3.8     89
4.2     80
3.9     78
3.7     75
4.1     66
3.6     61
4.3     49
3.4     48
3.5     48
4.4     45
3.0     45
4.5     44
3.2     33
4.6     31
1.0     31
3.3     29
4.7     21
3.1     16
2.5     15
2.0     13
2.6      7
2.7      7
Get      6
2.9      6
2.4      4
4.8      4
2.3      4
2.8      3
2.2      2
1.5      2
1.9      2
1.3      1
1.4      1
1.7      1
Name: ratings, dtype: int64

## Sorting

### sort values by column

In [21]:
pandacsv.sort_values(by='ratings', ascending=True)

name  main_category  \
680   TheYaYaCafe Bura na Mano Holi hai Matching Men...  kids' fashion   
1471  NASA Winter Sweater Cotton Full Sleeve Romper ...  kids' fashion   
770   Rsmart Non-Slip Cotton Floral Print Ribbon Clo...  kids' fashion   
748   SRCREATIONS Baby Girl's&Boys's Polycotton Prin...  kids' fashion   
364   Decent Apparel SunLook Girl's Dress Girls Ethn...  kids' fashion   
...                                                 ...            ...   
2200  6 To 12 Months / 1 Year Baby Boys & Baby Girls...  kids' fashion   
2201  My Bub Midi Frock | Printed Frock with Shorts|...  kids' fashion   
2203  Aswadh Cute Cartoon Animal Uni Kids Toddler So...  kids' fashion   
2204  B 4 Baby - Vest for Kids & Baby, Cotton Sando ...  kids' fashion   
2206  Aswadh Cute Cartoon Animal Uni Kids Fleece Sho...  kids' fashion   

      sub_category                                              image  \
680   Baby Fashion  https://m.media-amazon.com/images/I/812vBnuZT7...   
1471  Baby Fashion  https://m.media-amazon.com/images/I/61qFSmXkl7...   
770   Baby Fashion  https://m.media-amazon.com/images/W/IMAGERENDE...   
748   Baby Fashion  https://m.media-amazon.com/images/I/913AuToquY...   
364   Baby Fashion  https://m.media-amazon.com/images/I/51TN5Of09B...   
...            ...                                                ...   
2200  Baby Fashion  https://m.media-amazon.com/images/I/31lGk+vDiH...   
2201  Baby Fashion  https://m.media-amazon.com/images/I/617Ql8mWI+...   
2203  Baby Fashion  https://m.media-amazon.com/images/I/514rJK08XQ...   
2204  Baby Fashion  https://m.media-amazon.com/images/I/41NYCtwZiX...   
2206  Baby Fashion  https://m.media-amazon.com/images/I/61afR294Ri...   

                                                   link ratings no_of_ratings  \
680   https://www.amazon.in/TheYaYaCafe-Matching-Fam...     1.0             3   
1471  https://www.amazon.in/Winter-Sweater-Cotton-Sl...     1.0             1   
770   https://www.amazon.in/Rsmart-Non-Slip-Cotton-C...     1.0             1   
748   https://www.amazon.in/CREATIONS-COTTON-DRESS-M...     1.0             1   
364   https://www.amazon.in/Decent-Apparel-SunLook-D...     1.0             1   
...                                                 ...     ...           ...   
2200  https://www.amazon.in/Months-Girls-Cotton-Boot...     NaN           NaN   
2201  https://www.amazon.in/My-Bub-Colorful-Printed-...     NaN           NaN   
2203  https://www.amazon.in/Aswadh-Cartoon-Animal-To...     NaN           NaN   
2204  https://www.amazon.in/Baby-Cotton-Sando-Baniya...     NaN           NaN   
2206  https://www.amazon.in/Aswadh-Cartoon-Animal-Fl...     NaN           NaN   

     discount_price actual_price  
680            ₹349         ₹799  
1471         ₹1,299       ₹1,999  
770            ₹329         ₹549  
748            ₹220         ₹668  
364            ₹199         ₹999  
...             ...          ...  
2200           ₹230       ₹1,059  
2201            NaN         ₹499  
2203           ₹974       ₹1,860  
2204           ₹749         ₹999  
2206         ₹1,295       ₹2,473  

[2208 rows x 9 columns]

## Slicing
mengakses spesifik kolom / baris

### Mengakses spesifik kolom

In [22]:
pandacsv['name']

0           EIO New Born Baby Gift Set -13 Piecess (Pink)
1       exciting Lives - Vintage Motorbike Pocket Watc...
2       Jewel string by Bk arts Evil Eye Baby Nazaria ...
3                         Superminis Unisex-baby's Bootie
4       My Newborn Baby Mittens Caps Gloves and Booty ...
                              ...                        
2203    Aswadh Cute Cartoon Animal Uni Kids Toddler So...
2204    B 4 Baby - Vest for Kids & Baby, Cotton Sando ...
2205    KETKAR Baby Gloves Winter Warm Lovely Knitted ...
2206    Aswadh Cute Cartoon Animal Uni Kids Fleece Sho...
2207    Cuckoos New Born Baby Boys & Girls 100% Cotton...
Name: name, Length: 2208, dtype: object

In [23]:
pandacsv[['name', 'actual_price']]

name actual_price
0         EIO New Born Baby Gift Set -13 Piecess (Pink)       ₹1,499
1     exciting Lives - Vintage Motorbike Pocket Watc...         ₹699
2     Jewel string by Bk arts Evil Eye Baby Nazaria ...         ₹799
3                       Superminis Unisex-baby's Bootie         ₹899
4     My Newborn Baby Mittens Caps Gloves and Booty ...         ₹900
...                                                 ...          ...
2203  Aswadh Cute Cartoon Animal Uni Kids Toddler So...       ₹1,860
2204  B 4 Baby - Vest for Kids & Baby, Cotton Sando ...         ₹999
2205  KETKAR Baby Gloves Winter Warm Lovely Knitted ...         ₹299
2206  Aswadh Cute Cartoon Animal Uni Kids Fleece Sho...       ₹2,473
2207  Cuckoos New Born Baby Boys & Girls 100% Cotton...         ₹899

[2208 rows x 2 columns]

### Mengakses Spesifik Baris

In [25]:
pandacsv.iloc[10:20]

name  main_category  \
10  vaghbhatt baby hair band Multi-Colour Nylon La...  kids' fashion   
11  LuvLap 100% Cotton Baby Caps, Mittens and Boot...  kids' fashion   
12  Kids Fitness Tracker Watch for Girls Boys, Dig...  kids' fashion   
13  CHIU Chu-Chu Black Shoes with Double Strap for...  kids' fashion   
14  Wearville New Born Baby Boy's & Baby Girls jab...  kids' fashion   
15  DHRUVS COLLECTION Exclusive BIS Hallmarked 925...  kids' fashion   
16  Luv Lap Liquid Cleanser, Anti-Bacterial, Food ...  kids' fashion   
17  CUTEABLY unisex-baby Multicolored Booties - 6 ...  kids' fashion   
18  DKNY Unisex-Child Stassi Menia (Little Big Kid...  kids' fashion   
19  Li'l Cupid Regular Fit Hooded Fleece Sweatshir...  kids' fashion   

    sub_category                                              image  \
10  Baby Fashion  https://m.media-amazon.com/images/I/614nSuPIhW...   
11  Baby Fashion  https://m.media-amazon.com/images/I/71XG9Mly8U...   
12  Baby Fashion  https://m.media-amazon.com/images/I/61MMzlOW47...   
13  Baby Fashion  https://m.media-amazon.com/images/I/51cXHqiTi+...   
14  Baby Fashion  https://m.media-amazon.com/images/I/71KhPZuuWg...   
15  Baby Fashion  https://m.media-amazon.com/images/I/51E7H38r9z...   
16  Baby Fashion  https://m.media-amazon.com/images/I/71p5fCtiCn...   
17  Baby Fashion  https://m.media-amazon.com/images/I/61B-3c9Fy+...   
18  Baby Fashion  https://m.media-amazon.com/images/I/81DNROf3bV...   
19  Baby Fashion  https://m.media-amazon.com/images/I/51217Nw-26...   

                                                 link ratings no_of_ratings  \
10  https://www.amazon.in/vaghbhatt-Multi-Colour-H...     4.0           768   
11  https://www.amazon.in/LuvLap-Cotton-Mittens-Bo...     4.0         2,439   
12  https://www.amazon.in/Fitness-Tracker-Stopwatc...     4.1            13   
13  https://www.amazon.in/CHiU-Double-Booties-15-1...     4.0         7,280   
14  https://www.amazon.in/Wearville-Girls-Clothing...     4.2           940   
15  https://www.amazon.in/DHRUVS-COLLECTION-Exclus...     4.4            39   
16  https://www.amazon.in/Luvlap-Anti-Bacterial-Ac...     4.4         5,883   
17  https://www.amazon.in/Ninos-World-Cotton-Carto...     3.5           805   
18  https://www.amazon.in/DKNY-Unisex-Child-Stassi...     NaN           NaN   
19  https://www.amazon.in/Lil-Cupid-Regular-Hooded...     NaN           NaN   

   discount_price actual_price  
10           ₹299         ₹599  
11            NaN         ₹398  
12         ₹4,227       ₹9,900  
13           ₹298         ₹499  
14           ₹499       ₹1,299  
15           ₹467         ₹899  
16           ₹509         ₹669  
17           ₹279         ₹499  
18            NaN    ₹7,197.48  
19           ₹749       ₹1,249

In [27]:
pandacsv.iloc[10:]

name  main_category  \
10    vaghbhatt baby hair band Multi-Colour Nylon La...  kids' fashion   
11    LuvLap 100% Cotton Baby Caps, Mittens and Boot...  kids' fashion   
12    Kids Fitness Tracker Watch for Girls Boys, Dig...  kids' fashion   
13    CHIU Chu-Chu Black Shoes with Double Strap for...  kids' fashion   
14    Wearville New Born Baby Boy's & Baby Girls jab...  kids' fashion   
...                                                 ...            ...   
2203  Aswadh Cute Cartoon Animal Uni Kids Toddler So...  kids' fashion   
2204  B 4 Baby - Vest for Kids & Baby, Cotton Sando ...  kids' fashion   
2205  KETKAR Baby Gloves Winter Warm Lovely Knitted ...  kids' fashion   
2206  Aswadh Cute Cartoon Animal Uni Kids Fleece Sho...  kids' fashion   
2207  Cuckoos New Born Baby Boys & Girls 100% Cotton...  kids' fashion   

      sub_category                                              image  \
10    Baby Fashion  https://m.media-amazon.com/images/I/614nSuPIhW...   
11    Baby Fashion  https://m.media-amazon.com/images/I/71XG9Mly8U...   
12    Baby Fashion  https://m.media-amazon.com/images/I/61MMzlOW47...   
13    Baby Fashion  https://m.media-amazon.com/images/I/51cXHqiTi+...   
14    Baby Fashion  https://m.media-amazon.com/images/I/71KhPZuuWg...   
...            ...                                                ...   
2203  Baby Fashion  https://m.media-amazon.com/images/I/514rJK08XQ...   
2204  Baby Fashion  https://m.media-amazon.com/images/I/41NYCtwZiX...   
2205  Baby Fashion  https://m.media-amazon.com/images/I/81yuipbsxk...   
2206  Baby Fashion  https://m.media-amazon.com/images/I/61afR294Ri...   
2207  Baby Fashion  https://m.media-amazon.com/images/I/512K9zJxPh...   

                                                   link ratings no_of_ratings  \
10    https://www.amazon.in/vaghbhatt-Multi-Colour-H...     4.0           768   
11    https://www.amazon.in/LuvLap-Cotton-Mittens-Bo...     4.0         2,439   
12    https://www.amazon.in/Fitness-Tracker-Stopwatc...     4.1            13   
13    https://www.amazon.in/CHiU-Double-Booties-15-1...     4.0         7,280   
14    https://www.amazon.in/Wearville-Girls-Clothing...     4.2           940   
...                                                 ...     ...           ...   
2203  https://www.amazon.in/Aswadh-Cartoon-Animal-To...     NaN           NaN   
2204  https://www.amazon.in/Baby-Cotton-Sando-Baniya...     NaN           NaN   
2205  https://www.amazon.in/KETKAR-Knitted-Striped-M...     3.3             3   
2206  https://www.amazon.in/Aswadh-Cartoon-Animal-Fl...     NaN           NaN   
2207  https://www.amazon.in/Cuckoos-Girls-Cotton-Uni...     4.0             1   

     discount_price actual_price  
10             ₹299         ₹599  
11              NaN         ₹398  
12           ₹4,227       ₹9,900  
13             ₹298         ₹499  
14             ₹499       ₹1,299  
...             ...          ...  
2203           ₹974       ₹1,860  
2204           ₹749         ₹999  
2205            NaN         ₹299  
2206         ₹1,295       ₹2,473  
2207           ₹399         ₹899  

[2198 rows x 9 columns]

In [29]:
pandacsv.iloc[:20]

name  main_category  \
0       EIO New Born Baby Gift Set -13 Piecess (Pink)  kids' fashion   
1   exciting Lives - Vintage Motorbike Pocket Watc...  kids' fashion   
2   Jewel string by Bk arts Evil Eye Baby Nazaria ...  kids' fashion   
3                     Superminis Unisex-baby's Bootie  kids' fashion   
4   My Newborn Baby Mittens Caps Gloves and Booty ...  kids' fashion   
5   DHRUVS COLLECTION 92.5 Pure Silver Black Beads...  kids' fashion   
6   Bold N Elegant - Be Bold Inside & Elegant Outs...  kids' fashion   
7             NPRC Unisex Wool Cap (436_Multicolor_M)  kids' fashion   
8           SooWoo® Boy’s Anti-Skid Grip Cotton Socks  kids' fashion   
9   Coolz Kids Chu-Chu Sound Musical First Walking...  kids' fashion   
10  vaghbhatt baby hair band Multi-Colour Nylon La...  kids' fashion   
11  LuvLap 100% Cotton Baby Caps, Mittens and Boot...  kids' fashion   
12  Kids Fitness Tracker Watch for Girls Boys, Dig...  kids' fashion   
13  CHIU Chu-Chu Black Shoes with Double Strap for...  kids' fashion   
14  Wearville New Born Baby Boy's & Baby Girls jab...  kids' fashion   
15  DHRUVS COLLECTION Exclusive BIS Hallmarked 925...  kids' fashion   
16  Luv Lap Liquid Cleanser, Anti-Bacterial, Food ...  kids' fashion   
17  CUTEABLY unisex-baby Multicolored Booties - 6 ...  kids' fashion   
18  DKNY Unisex-Child Stassi Menia (Little Big Kid...  kids' fashion   
19  Li'l Cupid Regular Fit Hooded Fleece Sweatshir...  kids' fashion   

    sub_category                                              image  \
0   Baby Fashion  https://m.media-amazon.com/images/I/61X3IMp9dv...   
1   Baby Fashion  https://m.media-amazon.com/images/I/71f3rkWNAF...   
2   Baby Fashion  https://m.media-amazon.com/images/I/51Fc5epiN1...   
3   Baby Fashion  https://m.media-amazon.com/images/I/71MNqAzGnA...   
4   Baby Fashion  https://m.media-amazon.com/images/I/61CHRBjLTd...   
5   Baby Fashion  https://m.media-amazon.com/images/I/618KZ5Vlo0...   
6   Baby Fashion  https://m.media-amazon.com/images/I/512Fau61yG...   
7   Baby Fashion  https://m.media-amazon.com/images/I/A1hUaXW-Hm...   
8   Baby Fashion  https://m.media-amazon.com/images/I/81mJmWUJT+...   
9   Baby Fashion  https://m.media-amazon.com/images/I/810iMPEu00...   
10  Baby Fashion  https://m.media-amazon.com/images/I/614nSuPIhW...   
11  Baby Fashion  https://m.media-amazon.com/images/I/71XG9Mly8U...   
12  Baby Fashion  https://m.media-amazon.com/images/I/61MMzlOW47...   
13  Baby Fashion  https://m.media-amazon.com/images/I/51cXHqiTi+...   
14  Baby Fashion  https://m.media-amazon.com/images/I/71KhPZuuWg...   
15  Baby Fashion  https://m.media-amazon.com/images/I/51E7H38r9z...   
16  Baby Fashion  https://m.media-amazon.com/images/I/71p5fCtiCn...   
17  Baby Fashion  https://m.media-amazon.com/images/I/61B-3c9Fy+...   
18  Baby Fashion  https://m.media-amazon.com/images/I/81DNROf3bV...   
19  Baby Fashion  https://m.media-amazon.com/images/I/51217Nw-26...   

                                                 link ratings no_of_ratings  \
0   https://www.amazon.in/EIO-Kidi-Born-Babys-Piec...     4.2         2,958   
1   https://www.amazon.in/exciting-Lives-Motorbike...     4.2           161   
2   https://www.amazon.in/jewel-string-evil-eye-na...     4.2           962   
3   https://www.amazon.in/Superminis-Velvet-Bootie...     4.1         2,046   
4   https://www.amazon.in/My-NewBorn-Baby-Mitten-B...     3.8         5,601   
5   https://www.amazon.in/DHRUVS-COLLECTION-Sterli...     4.0           264   
6   https://www.amazon.in/Bold-Elegant-Outside-Kne...     3.9         4,050   
7   https://www.amazon.in/NPRC-Woollen-Monkey-Colo...     4.1             2   
8   https://www.amazon.in/SooWoo%C2%AE-Anti-Skid-C...     NaN           NaN   
9   https://www.amazon.in/Coolz-Chu-Chu-Musical-Wa...     4.2         5,914   
10  https://www.amazon.in/vaghbhatt-Multi-Colour-H...     4.0           768   
11  https://www.amazon.in/LuvLap-Cotton-Mittens-Bo...     4.0         2,439   
12  https://www.a

## Query / Filtering

### Query

In [32]:
pandacsv.query("ratings >= '2' and ratings <= '3'")

name  main_category  \
73    Neska Moda Months Baby Boys Synthetic Leather ...  kids' fashion   
78    Pooja Baby 100% Cotton Rompers/Sleepsuits/Jump...  kids' fashion   
94    DARSHRAJ JEWELLERS 925 Sterling Silver kada fo...  kids' fashion   
103   Traditional Silver & Black Nazariya Beads For ...  kids' fashion   
105                        Mozzo Baby boy & Girls Shoes  kids' fashion   
...                                                 ...            ...   
1738  Kidbee Baby Soft Super Cotton Hosiery Boys & B...  kids' fashion   
1830  Nino Bambino 100% Organic Cotton Printed Long ...  kids' fashion   
2041  HIYA Baby Boys/Girls Winter Wear Thermal Set W...  kids' fashion   
2052  BAYBEE Supersoft Cotton Baby Mittens, Booties ...  kids' fashion   
2143  Adorazone® Kid's Super-Warm Velvet Thermal Ank...  kids' fashion   

      sub_category                                              image  \
73    Baby Fashion  https://m.media-amazon.com/images/I/61ndWCatTz...   
78    Baby Fashion  https://m.media-amazon.com/images/I/51yCdftn5Z...   
94    Baby Fashion  https://m.media-amazon.com/images/I/71Vl8bdqsx...   
103   Baby Fashion  https://m.media-amazon.com/images/I/51vFo4NSCb...   
105   Baby Fashion  https://m.media-amazon.com/images/I/816L12+nRZ...   
...            ...                                                ...   
1738  Baby Fashion  https://m.media-amazon.com/images/I/61sGmKFleh...   
1830  Baby Fashion  https://m.media-amazon.com/images/W/IMAGERENDE...   
2041  Baby Fashion  https://m.media-amazon.com/images/I/61Fkst4EEJ...   
2052  Baby Fashion  https://m.media-amazon.com/images/I/714+yKbrOW...   
2143  Baby Fashion  https://m.media-amazon.com/images/I/61Rfs9loec...   

                                                   link ratings no_of_ratings  \
73    https://www.amazon.in/Neska-Moda-Synthetic-Lea...     2.5             8   
78    https://www.amazon.in/Cotton-Rompers-Sleepsuit...     2.5             3   
94    https://www.amazon.in/DARSHRAJ-Sterling-8-590g...     2.5             2   
103   https://www.amazon.in/Traditional-Silver-Black...     2.6             5   
105   https://www.amazon.in/Mozzo-Chu-Chu-Sound-Musi...     2.7             7   
...                                                 ...     ...           ...   
1738  https://www.amazon.in/Kidbee-Printed-Hosiery-C...     2.3             2   
1830  https://www.amazon.in/Nino-Bambino-Organic-Cot...     2.7             8   
2041  https://www.amazon.in/HIYA-Baby-Thermal-Set-Mu...     2.4             4   
2052  https://www.amazon.in/Baybee-Supersoft-Mittens...     2.5             3   
2143  https://www.amazon.in/Adorazone%C2%AE-Super-Wa...     2.4             4   

     discount_price actual_price  
73             ₹325         ₹599  
78             ₹499       ₹1,899  
94           ₹1,320       ₹2,999  
103            ₹199         ₹499  
105            ₹269         ₹599  
...             ...          ...  
1738           ₹380         ₹599  
1830        ₹759.60       ₹1,899  
2041           ₹449         ₹899  
2052           ₹831       ₹1,299  
2143           ₹399       ₹1,499  

[61 rows x 9 columns]

In [31]:
# hanya bisa 1 karena bukan label-based indexing
pandacsv.loc[pandacsv['ratings'] > '5']

name  main_category  \
22    Neska Moda 6 To 12 Months Baby Girls Synthetic...  kids' fashion   
53    WILDBIRDS Kids Girls Shoes for 12 Months to 10...  kids' fashion   
70    Neska Moda 0 To 6 Months Baby Boys & Girls Pac...  kids' fashion   
74    superbottoms Top & Shorts Set | Evrydaywear| C...  kids' fashion   
81                        KingSwagger Sneakers for Kids  kids' fashion   
...                                                 ...            ...   
2172            Gini & Jony Boys Round Neck Printed Tee  kids' fashion   
2175  MINI KLUB Baby Boy's Cotton Cap (95IEBCP569_Of...  kids' fashion   
2187  Neska Moda Baby Boys and Girls Pack of 4 Cotto...  kids' fashion   
2195  Nutshell Baby Clothing Set for Boys & Girls | ...  kids' fashion   
2202  ARIF Collection Baby Girls Calf Length Casual ...  kids' fashion   

      sub_category                                              image  \
22    Baby Fashion  https://m.media-amazon.com/images/I/719s4QLtRF...   
53    Baby Fashion  https://m.media-amazon.com/images/I/614x5PKumZ...   
70    Baby Fashion  https://m.media-amazon.com/images/I/81t-vX6X-A...   
74    Baby Fashion  https://m.media-amazon.com/images/I/51rj-BhS+V...   
81    Baby Fashion  https://m.media-amazon.com/images/I/61rrbvVOXn...   
...            ...                                                ...   
2172  Baby Fashion  https://m.media-amazon.com/images/I/5197kxNNZD...   
2175  Baby Fashion  https://m.media-amazon.com/images/I/71-smTzBap...   
2187  Baby Fashion  https://m.media-amazon.com/images/I/71AnbV6I49...   
2195  Baby Fashion  https://m.media-amazon.com/images/I/71a-t3PeHj...   
2202  Baby Fashion  https://m.media-amazon.com/images/I/715ptsgVLE...   

                                                   link ratings no_of_ratings  \
22    https://www.amazon.in/Neska-Moda-Synthetic-Lea...     5.0             4   
53    https://www.amazon.in/WILDBIRDS-Kids-Shoes-Chi...     5.0             2   
70    https://www.amazon.in/Neska-Moda-Months-Bootie...     5.0             1   
74    https://www.amazon.in/SuperBottoms-Evrydaywear...     5.0             2   
81    https://www.amazon.in/KingSwagger-Kids-Sneaker...     5.0             4   
...                                                 ...     ...           ...   
2172  https://www.amazon.in/Gini-Jony-Printed-Yellow...     5.0             1   
2175  https://www.amazon.in/Mini-Klub-Cotton-95IEBCP...     5.0             1   
2187  https://www.amazon.in/Neska-Moda-Girls-Multico...     5.0             1   
2195  https://www.amazon.in/Nutshell-Clothing-Regula...     5.0             2   
2202  https://www.amazon.in/Collection-Length-Flawso...     5.0             1   

     discount_price actual_price  
22             ₹280         ₹599  
53             ₹399         ₹899  
70             ₹369         ₹699  
74             ₹229         ₹449  
81             ₹384         ₹899  
...             ...          ...  
2172        ₹244.47         ₹499  
2175           ₹217         ₹249  
2187           ₹349         ₹899  
2195           ₹619       ₹1,499  
2202           ₹329         ₹899  

[156 rows x 9 columns]

Pertama-tama mari convert value pada kolom actual_price terlebih dahulu

In [39]:
pandacsv['actual_price_int'] = pandacsv['actual_price'].str.replace('₹', '').str.replace(',', '')
pandacsv['actual_price_int'] = pandacsv['actual_price_int'].astype(float)

pandacsv

name  main_category  \
0         EIO New Born Baby Gift Set -13 Piecess (Pink)  kids' fashion   
1     exciting Lives - Vintage Motorbike Pocket Watc...  kids' fashion   
2     Jewel string by Bk arts Evil Eye Baby Nazaria ...  kids' fashion   
3                       Superminis Unisex-baby's Bootie  kids' fashion   
4     My Newborn Baby Mittens Caps Gloves and Booty ...  kids' fashion   
...                                                 ...            ...   
2203  Aswadh Cute Cartoon Animal Uni Kids Toddler So...  kids' fashion   
2204  B 4 Baby - Vest for Kids & Baby, Cotton Sando ...  kids' fashion   
2205  KETKAR Baby Gloves Winter Warm Lovely Knitted ...  kids' fashion   
2206  Aswadh Cute Cartoon Animal Uni Kids Fleece Sho...  kids' fashion   
2207  Cuckoos New Born Baby Boys & Girls 100% Cotton...  kids' fashion   

      sub_category                                              image  \
0     Baby Fashion  https://m.media-amazon.com/images/I/61X3IMp9dv...   
1     Baby Fashion  https://m.media-amazon.com/images/I/71f3rkWNAF...   
2     Baby Fashion  https://m.media-amazon.com/images/I/51Fc5epiN1...   
3     Baby Fashion  https://m.media-amazon.com/images/I/71MNqAzGnA...   
4     Baby Fashion  https://m.media-amazon.com/images/I/61CHRBjLTd...   
...            ...                                                ...   
2203  Baby Fashion  https://m.media-amazon.com/images/I/514rJK08XQ...   
2204  Baby Fashion  https://m.media-amazon.com/images/I/41NYCtwZiX...   
2205  Baby Fashion  https://m.media-amazon.com/images/I/81yuipbsxk...   
2206  Baby Fashion  https://m.media-amazon.com/images/I/61afR294Ri...   
2207  Baby Fashion  https://m.media-amazon.com/images/I/512K9zJxPh...   

                                                   link ratings no_of_ratings  \
0     https://www.amazon.in/EIO-Kidi-Born-Babys-Piec...     4.2         2,958   
1     https://www.amazon.in/exciting-Lives-Motorbike...     4.2           161   
2     https://www.amazon.in/jewel-string-evil-eye-na...     4.2           962   
3     https://www.amazon.in/Superminis-Velvet-Bootie...     4.1         2,046   
4     https://www.amazon.in/My-NewBorn-Baby-Mitten-B...     3.8         5,601   
...                                                 ...     ...           ...   
2203  https://www.amazon.in/Aswadh-Cartoon-Animal-To...     NaN           NaN   
2204  https://www.amazon.in/Baby-Cotton-Sando-Baniya...     NaN           NaN   
2205  https://www.amazon.in/KETKAR-Knitted-Striped-M...     3.3             3   
2206  https://www.amazon.in/Aswadh-Cartoon-Animal-Fl...     NaN           NaN   
2207  https://www.amazon.in/Cuckoos-Girls-Cotton-Uni...     4.0             1   

     discount_price actual_price  actual_price_int  
0              ₹649       ₹1,499            1499.0  
1              ₹485         ₹699             699.0  
2              ₹249         ₹799             799.0  
3              ₹350         ₹899             899.0  
4              ₹299         ₹900             900.0  
...             ...          ...               ...  
2203           ₹974       ₹1,860            1860.0  
2204           ₹749         ₹999             999.0  
2205            NaN         ₹299             299.0  
2206         ₹1,295       ₹2,473            2473.0  
2207           ₹399         ₹899             899.0  

[2208 rows x 10 columns]

In [42]:
del pandacsv['actual_price']

KeyError: 'actual_price'

In [41]:
pandacsv

name  main_category  \
0         EIO New Born Baby Gift Set -13 Piecess (Pink)  kids' fashion   
1     exciting Lives - Vintage Motorbike Pocket Watc...  kids' fashion   
2     Jewel string by Bk arts Evil Eye Baby Nazaria ...  kids' fashion   
3                       Superminis Unisex-baby's Bootie  kids' fashion   
4     My Newborn Baby Mittens Caps Gloves and Booty ...  kids' fashion   
...                                                 ...            ...   
2203  Aswadh Cute Cartoon Animal Uni Kids Toddler So...  kids' fashion   
2204  B 4 Baby - Vest for Kids & Baby, Cotton Sando ...  kids' fashion   
2205  KETKAR Baby Gloves Winter Warm Lovely Knitted ...  kids' fashion   
2206  Aswadh Cute Cartoon Animal Uni Kids Fleece Sho...  kids' fashion   
2207  Cuckoos New Born Baby Boys & Girls 100% Cotton...  kids' fashion   

      sub_category                                              image  \
0     Baby Fashion  https://m.media-amazon.com/images/I/61X3IMp9dv...   
1     Baby Fashion  https://m.media-amazon.com/images/I/71f3rkWNAF...   
2     Baby Fashion  https://m.media-amazon.com/images/I/51Fc5epiN1...   
3     Baby Fashion  https://m.media-amazon.com/images/I/71MNqAzGnA...   
4     Baby Fashion  https://m.media-amazon.com/images/I/61CHRBjLTd...   
...            ...                                                ...   
2203  Baby Fashion  https://m.media-amazon.com/images/I/514rJK08XQ...   
2204  Baby Fashion  https://m.media-amazon.com/images/I/41NYCtwZiX...   
2205  Baby Fashion  https://m.media-amazon.com/images/I/81yuipbsxk...   
2206  Baby Fashion  https://m.media-amazon.com/images/I/61afR294Ri...   
2207  Baby Fashion  https://m.media-amazon.com/images/I/512K9zJxPh...   

                                                   link ratings no_of_ratings  \
0     https://www.amazon.in/EIO-Kidi-Born-Babys-Piec...     4.2         2,958   
1     https://www.amazon.in/exciting-Lives-Motorbike...     4.2           161   
2     https://www.amazon.in/jewel-string-evil-eye-na...     4.2           962   
3     https://www.amazon.in/Superminis-Velvet-Bootie...     4.1         2,046   
4     https://www.amazon.in/My-NewBorn-Baby-Mitten-B...     3.8         5,601   
...                                                 ...     ...           ...   
2203  https://www.amazon.in/Aswadh-Cartoon-Animal-To...     NaN           NaN   
2204  https://www.amazon.in/Baby-Cotton-Sando-Baniya...     NaN           NaN   
2205  https://www.amazon.in/KETKAR-Knitted-Striped-M...     3.3             3   
2206  https://www.amazon.in/Aswadh-Cartoon-Animal-Fl...     NaN           NaN   
2207  https://www.amazon.in/Cuckoos-Girls-Cotton-Uni...     4.0             1   

     discount_price  actual_price_int  
0              ₹649            1499.0  
1              ₹485             699.0  
2              ₹249             799.0  
3              ₹350             899.0  
4              ₹299             900.0  
...             ...               ...  
2203           ₹974            1860.0  
2204           ₹749             999.0  
2205            NaN             299.0  
2206         ₹1,295            2473.0  
2207           ₹399             899.0  

[2208 rows x 9 columns]

In [43]:
pandacsv['actual_price_int'].mean()

1285.6823977695165

In [44]:
pandacsv.groupby('ratings')['name'].count()

ratings
1.0     31
1.3      1
1.4      1
1.5      2
1.7      1
1.9      2
2.0     13
2.2      2
2.3      4
2.4      4
2.5     15
2.6      7
2.7      7
2.8      3
2.9      6
3.0     45
3.1     16
3.2     33
3.3     29
3.4     48
3.5     48
3.6     61
3.7     75
3.8     89
3.9     78
4.0    145
4.1     66
4.2     80
4.3     49
4.4     45
4.5     44
4.6     31
4.7     21
4.8      4
5.0    150
Get      6
Name: name, dtype: int64